src/python/pants/backend/tools/semgrep/subsystem.py

```diff
@@ -51,6 +57,20 @@ class SemgrepSubsystem(PythonToolBase):
     register_lockfile = True
     default_lockfile_resource = ("pants.backend.tools.semgrep", "semgrep.lock")
 
+    config_paths = StrListOption(
+        default=[".semgrep"],
```

Comment:
Haha, I was going to mention this in the pr but must've overwritten it. 

I didn't do that precisely for the same reason you mentioned - didn't want to add too much at once and then deal with the globs. I was thinking of first just allowing customizing the dir, and then allowing files in the next pr. 

I even had this named `config_dir` at first, and then changed the name so that allowing files could just be a simple release note rather than a breaking change.

I can add it to this pr itself.

Reply:
Updated the way this config works

Reply:
Thanks for updating this; this looks like a good incremental improvement, and I'm definitely sympathic to trying to keep the changes smaller.

> didn't want to add too much at once

For the next steps, if you keep working on this, I'd still be interested in moving most of the path/globbing logic into the option in `pants.toml`.

For example, evolving to  `config_paths` that takes a list of paths/globs (or maybe inspired by `root_patterns` https://www.pantsbuild.org/2.21/docs/using-pants/key-concepts/source-roots), instead of the single `config_name`.

Reply:
Not necessarily a big deal, but I wanted to keep it restricted to `config_name` since it allows us to treat the repo structure more uniformly. I also wonder how many people configure semgrep in completely different ways for each project.

I don't mind following up with that though. Probably at the same time as deprecating defaults, since that'll make things easier too.